Issue #3411: PhysicalRangeJoin

hawkfish opened the pull request:
Refactor IEJoin and PiecewiseMergeJoin to use common code for sorted table construction and other tasks.

Commit:
Issue #3411: Merge selection simplification

There is no need for all the strange tricks used
to modify the left SV in place. These changes make
it so we can unify the two SliceSortedPayload versions.

Commit:
Issue #3411: Derive Merge from Range

Convert PhysicalPiecewiseMergeJoin to be
a subclass of PhysicalRangeJoin and
start using the static members.

Commit:
Issue #3411: Derive IEJoin from Range

Convert PhysicalIEJoin to be
a subclass of PhysicalRangeJoin and
start using the static members.

Commit:
Issue #3411: LocalSortedTable in Merge

Use a new common class for local sorted table sinks
in PhysicalPiecewiseMergeJoin.

Commit:
Issue #3411: LocalSortedTable in IEJoin

Use LocalSortedTable in IEJoinLocalState.

Commit:
Issue #3411: Refactor GlobalSortedTable Logic

Use in IEJoin.

Commit:
Issue #3411: GlobalSortedTable in Merge

Also move null merging to LocalSortedTable.

Commit:
Issue #3411: Refactor Sorting Events

Use with PhysicalIEJoin.

Commit:
Issue #3411: Refactor Sorting Events

Use with PhysicalPiecewiseMergeJoin.

Commit:
Merge branch 'master' into hawkfish-range-join

Commit:
Issue #3411: Refactor IEJoin SelectJoinTail

Switch to the common code.

lnkuiper commented on src/execution/operator/join/physical_range_join.cpp:
```diff
@@ -0,0 +1,357 @@
+	memset(found_match.get(), 0, sizeof(bool) * Count());
+}
+
+void PhysicalRangeJoin::GlobalSortedTable::Print() {
```
Could this Print function be moved into the GlobalSortState? It seems like it's more generally applicable than just the PhysicalRangeJoin

lnkuiper reviewed (changes requested):
Nice work! Great idea merging these two since they are so similar. One minor nitpick ...

Commit:
Issue #3411: Move GlobalSortedTable::Print

Since it only depends on the GlobalSortState,
it makes sense to make this a method of that class.

Mytherin commented:
Thanks!
